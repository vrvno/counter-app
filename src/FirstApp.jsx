import React from 'react';
import PropTypes from 'prop-types';

//default props deben ir en las propiedades (defaultProps no longer supported)
export const FirstApp = ({
	title = 'no hay titulo',
	subTitle = 'no hay subtitulo',
	nombre = 'no hay nombre',
}) => {
	//console.log(props);
	return (
		<>
			<div data-testid='test-title'>{title}</div>
			{/*<code>{JSON.stringify(newMessage)}</code>*/}
			<p>{subTitle}</p>
			<p>{nombre}</p>
		</>
	);
};

FirstApp.propTypes = {
	title: PropTypes.string.isRequired,
	subTitle: PropTypes.number.isRequired,
};
