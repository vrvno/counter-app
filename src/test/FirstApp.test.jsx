import React from 'react';
import { render } from '@testing-library/react';
import { FirstApp } from '../FirstApp';

describe('Pruebas en <FirstApp/>', () => {
	// test('debe hacer match con el snapshot', () => {
	// 	const title = 'uHola, Soy Goku';
	// 	const { container } = render(<FirstApp title={title} />);

	// 	expect(container).toMatchSnapshot();
	// });

	test('debe de mostrar el titulo en un h1', () => {
		const title = 'Hola, Soy Goku';
		const { container, getByText, getByTestId } = render(
			<FirstApp title={title} />
		);
		expect(getByText(title)).toBeTruthy();

		// const h1 = container.querySelector('h1');
		// expect(h1.innerHTML).toContain(title);
		expect(getByTestId('test-title').innerHTML).toContain('Hola, Soy Goku');
	});

	test('debe de mostrar subtitulo', () => {
		const title = 'Hola, Soy Goku';
		const subTitle = 'Soy un subt√≠tulo';
		const { getAllByText } = render(
			<FirstApp title={title} subTitle={subTitle} />
		);

		expect(getAllByText(subTitle).length).toBe(1);
	});
});
