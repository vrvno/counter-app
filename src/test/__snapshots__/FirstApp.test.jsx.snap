// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, Soy Goku
  </h1>
  <p>
    no hay subtitulo
  </p>
  <p>
    no hay nombre
  </p>
</div>
`;
